spec/requests/session_controller_spec.rb

```diff
@@ -2260,4 +2260,195 @@ def post_login
       end
     end
   end
+
+  describe '#second_factor_auth_show' do
+    let!(:user_second_factor) { Fabricate(:user_second_factor_totp, user: user) }
+
+    before do
+      sign_in(user)
+    end
+
+    it 'returns 404 if there is no challenge for the given nonce' do
+      get "/session/2fa.json", params: { nonce: 'asdasdsadsad' }
+      expect(response.status).to eq(404)
+      expect(response.parsed_body["error"]).to eq(I18n.t("second_factor_auth.challenge_not_found"))
+    end
+
+    it 'returns 404 if the nonce does not match the challenge nonce' do
+      post "/session/2fa/test-action"
+      expect(response.status).to eq(403)
+      expect(response.parsed_body["second_factor_challenge_nonce"]).to be_present
```

Comment:
Just curious, what is the purpose of this request in relation to the request below? Should we perhaps split this test up into two?

Reply:
This test case is for the `#second_factor_auth_show` action which requires a 2fa challenge to exist in the session, otherwise it'd just respond with a 404. To be able to test that action, I need something to set up a challenge in the session for me, and that's what the post request to `/session/2fa/test-action` does. The assertions for the response/status code of that helper/test action are not really useful, so I've removed them.